src/bit_iterator.rs

```diff
@@ -8,54 +8,47 @@
 //! limbs
 use core::mem;
 
-macro_rules! bit_iterator {
-    ($sty : ty, $name : ident) => {
-        #[derive(Debug, Clone, Copy)]
-        pub struct $name<E> {
-            // scalar is the slice of integers that wish to iterate over
-            scalar: E,
-            // num_of_total_bits represents the sum of all of the bits of each
-            // integer If we have 2 u32s then the total number of bits will
-            // be 32 * 2 = 64 bits
-            num_of_total_bits: usize,
-            // bit_len represents the bit length of each integer.
-            // If we have a slice of u32s, then bit_len will be 32
-            bit_len: usize,
-        }
+pub struct BitIterator8<E> {
```

Comment:
You didn't derive the `Debug`, `Clone` and `Copy` traits. Can you still include those?

Reply:
done!